src/rsz/src/SwapArithModules.hh

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2025-2025, The OpenROAD Authors
+
+#pragma once
+
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+#include "db_sta/dbNetwork.hh"
+#include "db_sta/dbSta.hh"
+#include "rsz/Resizer.hh"
+#include "sta/MinMax.hh"
+#include "sta/StaState.hh"
+#include "utl/Logger.h"
+
+namespace odb {
+class dbDatabase;
+class dbInst;
+}  // namespace odb
+
+namespace sta {
+class dbSta;
+class LibertyCell;
+class PathEnd;
+class Pin;
+class PinSet;
+}  // namespace sta
+
+namespace utl {
+class Logger;
+}
+
+namespace rsz {
+
+class Resizer;
+
+using odb::dbModInst;
+using sta::Path;
+using std::unordered_set;
+using utl::Logger;
+
+class SwapArithModules : public sta::dbStaState
```

Comment:
What motivates the virtual base class here as there is only one derived class.  Do you expect more in the future?  Are these all methods a client would use (eg collectArithInstsOnPath)?  Currently on replaceArithModules is used from what I see.